src/main/java/com/google/cloud/discotoproto3converter/proto3/DocumentToProtoConverter.java

```diff
@@ -306,37 +305,36 @@ private void readResources(Document document) {
         String inputDescription = getInputMessageDescription(grpcServiceName, methodname);
         Message input = new Message(requestName, false, false, inputDescription);
         String httpOptionPath = method.flatPath();
-        List<String> methodSignatureParams = new ArrayList<>();
-
-        for (Schema requiredParam : method.requiredParams().values()) {
-          Field requiredField = schemaToField(requiredParam);
-          Option fieldBehaviorOption = new Option("google.api.field_behavior");
-          fieldBehaviorOption.getProperties().put("", "REQUIRED");
-          requiredField.getOptions().add(fieldBehaviorOption);
-          input.getFields().add(requiredField);
-          methodSignatureParams.add(requiredField.getName());
+        Map<String, String> methodSignatureParamNames = new LinkedHashMap<>();
+        for (String requiredParamName : method.requiredParamNames()) {
+          methodSignatureParamNames.put(requiredParamName, null);
```

Comment:
And just to confirm: the only reason you're pre-populating the `LinkedhashMap` with empty values is so you can preserve the order of the parameters (already determined) regardless of how you encounter them below? (sg, just checking)

Reply:
Yes, it is correct.